audbcards/core/dataset.py

```diff
@@ -274,6 +292,283 @@ def version_link(self) -> str:
+
+    @property
+    def scheme_info(self) -> dict:
+
+        db = self.header
+        scheme_info = {}
+
+        if len(db.schemes) > 0:
+            has_minimums = any(
+                [db.schemes[s].minimum is not None for s in db.schemes]
+            )
+            has_maximums = any(
+                [db.schemes[s].maximum is not None for s in db.schemes]
+            )
+            has_labels = any(
+                [db.schemes[s].labels is not None for s in db.schemes]
+            )
+            has_mappings = any(
+                [
+                    isinstance(db.schemes[s].labels, (str, dict))
+                    for s in db.schemes
+                ]
+            )
+
+        columns = ['ID', 'Dtype']
+        header_line = '   :header: ID,Dtype'
+        if has_minimums:
+            header_line += ',Min'
+            columns.append('Min')
+        if has_maximums:
+            header_line += ',Max'
+            columns.append('Max')
+        if has_labels:
+            header_line += ',Labels'
+            columns.append('Labels')
+        if has_mappings:
+            header_line += ',Mappings'
+            columns.append('Mappings')
+
+        scheme_info['header_line'] = header_line
+        scheme_info['has_minimums'] = has_minimums
+        scheme_info['has_maximums'] = has_maximums
+        scheme_info['has_labels'] = has_labels
+        scheme_info['has_mappings'] = has_mappings
+
+        scheme_info['columns'] = columns
+        return scheme_info
+
+    @property
+    def dataset_schemes(self) -> list:
+
+        db = self.header
+        dataset_schemes = []
+        for scheme_id in db.schemes:
+            dataset_scheme = self._scheme_to_list(scheme_id)
+            dataset_schemes.append(dataset_scheme)
+
+        cols = self.scheme_info['columns']
+        data = pd.DataFrame.from_dict(dataset_schemes)[cols]
+        filter = data.applymap(lambda d: d == [])
+        data.mask(filter, other='', inplace=True)
+        scheme_data = np.array(data).tolist()
+        return scheme_data
+
+
+class Datacard(object):
+
+    def __init__(self, dataset: Dataset):
+
+        self._dataset = dataset
+
+    @property
+    def props(self) -> dict:
+        props = self._dataset.properties()
+        return props
+
+    def _render_template(self):
+
+        t_dir = os.path.join(os.path.dirname(__file__), 'templates')
+        environment = jinja2.Environment(loader=jinja2.FileSystemLoader(t_dir),
+                                         trim_blocks=True)
+        environment.filters.update(zip=zip,
+                                   tw=self._trim_trailing_whitespace,
+                                   )
+        template = environment.get_template("datacard.j2")
+        content = template.render(self.props)
+        return content
+
+    @staticmethod
+    def _trim_trailing_whitespace(x: list):
+        """J2 filter to get rid or trailing empty table entries within a row.
+
+        Trims last entry if present.
```

Comment:
What does "if present" means here?

Why do we need to remove empty columns?

Reply:
Honestly I forgot. It is related to the fact that the datasets differ in the attributes they have, e.g. one dataset might have `ID, Dtype, Min, Max, Labels, Mappings` and the next one misses one or more. Very frequently `Mapping`  is missing. It is entirely possible that there is the need for more test cases.

Reply:
The default page is compiled, as is the emodb car  in https://github.com/audeering/datasets/pull/8.  So should we discuss remaining edge cases in a separate issue / MR?

Reply:
Yes, I would open issues and add tests for edge cases after this pull request.

Reply:
I tested compiling a data card for `emodb` with and without `_trim_trailing_whitespace()` and couldn't find any difference.

I would propose that we stay with it for now, but I also created https://github.com/audeering/audbcards/issues/21 to check if it is really needed.